contrib/scalar/scalar.c

```diff
@@ -25,8 +25,9 @@ static int is_unattended(void) {
  */
 static void strbuf_parentdir(struct strbuf *buf)
 {
-	char *path_sep = find_last_dir_sep(buf->buf);
-	strbuf_setlen(buf, path_sep ? path_sep - buf->buf : 0);
+	size_t offset = offset_1st_component(buf->buf);
+	char *path_sep = find_last_dir_sep(buf->buf + offset);
+	strbuf_setlen(buf, path_sep ? path_sep - buf->buf : offset);
```

Comment:
I still wonder whether we should indicate in an `int` return value whether the path was modified or not, to avoid infinite loops in the callers.

Reply:
Done - it really helped with simplifying some of the conditions in `find_enlistment(...)`!